src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -188,6 +195,101 @@ public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerial
             return default;
         }
 
+        private static bool IsNumericString(ref Utf8JsonReader reader)
+        {
+            ReadOnlySpan<byte> span = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
```

Comment:
Worst-case scenario this is going to incur one added allocation in the hot path. Would it make sense to integrate the check in the `TryParseEnumCore` methods instead?

Reply:
Check logic is moved into `TryParseEnumCore()`. 
For ROS versioned `TryParseEnumCore()`, check happens after extracting utf-16 based string allocation from pool (existing one), please check whether proper or not.